yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -608,6 +612,18 @@ private[spark] class ApplicationMaster(
     }
   }
 
+  /** set  "SPARK_SHUFFLE_TOKEN" before registerAM
+    * @return
+    */
+  private def initJobCredentialsAndUGI() = {
```

Comment:
I mentioned this before but I don't believe this should be necessary. `Client` always runs before this and should be propagating the credentials to the AM. In fact, if somehow the secret generated by `Client` is not showing up here, you could have weird things happening in client mode, since the driver and the AM will have different secrets.